src/components/Calculator.js

```diff
@@ -1,44 +1,65 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import Calculate from '../logic/calculate';
 
 // DigitButton component with prop validation
-function DigitButton({ digit }) {
+function DigitButton({ digit, onClick }) {
   return (
-    <button type="button" className="digit-button">{ digit }</button>
+    <button type="button" className="digit-button" onClick={() => onClick(digit)}>{ digit }</button>
   );
 }
 DigitButton.propTypes = {
-  digit: PropTypes.oneOfType(PropTypes.string, PropTypes.number).isRequired,
+  digit: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  onClick: PropTypes.func.isRequired,
 };
 
 // OpertorButton component with prop validation
-function OperatorButton({ operator }) {
+function OperatorButton({ operator, onClick }) {
   return (
-    <button type="button" className="operator-button">{ operator }</button>
+    <button type="button" className="operator-button" onClick={() => onClick(operator)}>{ operator }</button>
   );
 }
 OperatorButton.propTypes = {
   operator: PropTypes.string.isRequired,
+  onClick: PropTypes.func.isRequired,
 };
```

Comment:
- **[OPTIONAL]** I would suggest you move these components to their own files, instead of lumping all components into one file.

Reply:
@DammyShittu Thanks, I will consider moving them to a different file.